### PR TITLE
Fix #25 - New game flow

### LINKED ISSUE #25: "New game" flow
The current "new game" flow needs to be improved.  How it should work:
- When the game is launched, play should **not** start immediately.  Instead a "ready to play" message should be displayed.
- Starting the game should be via a keystroke, not via the mouse.
- When the game ends, starting the next game should also be via keystroke.

### PR DESCRIPTION
**Pull request by mgoimam**
- Start with no game set and a welcome screen.
- New command to start a new game [Enter/Return on desktops, Start on
  gamepads].
- Add null check in UDPRenderer since game can now be null.
- Simplified the logic in the AggregateGameInput to make adding new
  commands a bit less cumbersome.

### CODE REVIEW

File: AppletRenderer.pde [hunk 1]
```diff
@@ -26,14 +20,24 @@ class AppletRenderer implements ControlListener {
...
+
+      textAlign(CENTER, BOTTOM);
+      textSize(18);
+      text("(press enter to begin)", width/2, height/2);
+
+      textSize(25);
+      if (currentGame == null) {
+        text("READY TO PLAY", width/2, height/2 - 35);
+      } else if (currentGame.isGameOver()) {
+        text("GAME OVER\nSCORE: " + currentGame.getScore(), width/2, height/2 - 50);
```
Comment (scjody) [inline AppletRenderer.pde]: Can you add the lines count here too please?

File: UDPRenderer.pde [hunk 2]
```diff
@@ -15,6 +15,8 @@ class UDPRenderer {
   }
   
   void renderGameState(TetrisGame currentGame) {
+    if (currentGame == null) return;
+
```
Comment (scjody) [inline UDPRenderer.pde:19]: Can you actually send an emptyGrid() in this case too?

**Comment by scjody** [comment]
Also can you send an emptyGrid() on game quit?

**Comment by mgoimam** [comment]
I'm only clearing the UDP renderer when the program is quit using the CTRL+Q shortcut. It should be possible to hook onto the application exit otherwise, but I'd need to research that a bit more. Also, I don't think doing it via the Processing app will ever be 100% reliable. If the application ever crashes or hangs, we still have a timeout on the server side that shuts off the valves, right?

**Comment by scjody** [comment]
Yeah, CTRL-Q is what I meant.  You're right, it's not actually necessary, it just means we'll turn off the fire a tiny bit sooner...

**Comment by scjody** [comment]
Great, :shipit: !

**Pull request merged**